source/auth.c

```diff
@@ -584,7 +584,6 @@ static napi_value s_creds_provider_new_x509(napi_env env, const struct aws_napi_
  **********************************************************************************************************************/
 
 struct signer_sign_request_state {
-    napi_ref node_request;
     struct aws_http_message *request;
```

Comment:
Simplification/Theoretical Fix: Just keep refcount on the C request. Don't bother storing pointer to the node_request.

Previously, we kept a refcount on the node_request, and just had a pointer to the C request (without doing any refcounting). That kiiinda works for the aws_sign_request() path, but the verify_sigv4a_signing() path forgot to refcount the node_request and just did a pointer to the C request which is SKETCHY.